bundles/framework/admin-layerrights/Flyout.js

```diff
@@ -527,29 +525,28 @@ Oskari.clazz.define('Oskari.framework.bundle.admin-layerrights.Flyout',
          * @param {String} selectedId
          */
         makeExternalIdsSelect: function (result, externalType, selectedId) {
-            var externalIdSelect = jQuery(this.container).find('select.admin-layerrights-role'),
-                optionEl,
-                d,
-                rightsLoc = this.instance._localization.rights;
+            var externalIdSelect = jQuery(this.container).find('select.admin-layerrights-role');
+            var rightsLoc = this.instance._localization.rights;
 
             externalIdSelect.html('');
-            if (externalType !== '0') {
+            if (externalType === '0') {
+                return;
+            }
+            var optionEl = document.createElement('option');
+            optionEl.value = '0';
+            optionEl.textContent = '-- ' + rightsLoc.selectValue + ' --';
+            if ('' + selectedId === '0') {
```

Comment:
It's safe to make the comparison with the number `0`. The method is currently only called with number argument as `selectedId` (see line 259)

Reply:
Good catch. Removed selectedId and externalType from params as they are basically constants.